src/pages/index.js

```diff
@@ -90,44 +90,93 @@ const profileEditPopup = new PopupWithForm(
 profileEditPopup.setEventListeners();
 
 // Profile Edit Popup Button
+const profileEditButton = document.querySelector("#profile-edit-button");
+
 profileEditButton.addEventListener("click", () => {
-  const userInfoData = userInfo.getUserInfo();
+  if (userInfoInstance) {
+    const userInfoData = userInfoInstance.getUserInfo();
 
-  profileTitleInput.value = userInfoData.name;
-  profileDescriptionInput.value = userInfoData.job;
+    profileTitleInput.value = userInfoData.name;
+    profileDescriptionInput.value = userInfoData.job;
 
-  profileEditPopup.open();
+    profileFormValidator.resetValidation();
+    profileEditPopup.open();
+  }
 });
 
-// Profile Changes Save
 function handleProfileSaveSubmit(inputValues) {
-  userInfo.setUserInfo({
+  const saveButton = document.getElementById("profile-modal-save-button");
+
+  const newInfo = {
     name: inputValues.title,
-    job: inputValues.description,
-  });
+    about: inputValues.description,
+    avatar: avatarImg.src,
+  };
+
+  saveButton.textContent = "Saving...";
+
+  apiUpdateProfileInfo(newInfo)
+    .then(() => {
+      userInfoInstance.setUserInfo({
+        name: inputValues.title,
+        job: inputValues.description,
+        avatar: avatarImg.src,
+      });
+    })
+    .catch((error) => {
+      console.error(error);
+      throw error;
```

Comment:
Please, delete `throw error` from blocks `catch` because you need to catch errors there rather than throw them away again